features/grpc/exporter/src/test/java/org/opennms/features/grpc/exporter/NmsInventoryServiceSyncIT.java

```diff
@@ -0,0 +1,730 @@
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class NmsInventoryServiceSyncIT implements TemporaryDatabaseAware<MockDatabase> {
+
+    private static final int PORT = 50051;
+    private static final String TENANT_ID = "opennms-prime";
+    private static final String HOST_NAME = "localhost";
+
+    private Server server;
+    private NmsInventoryGrpcClient client;
+
+    @Autowired
+    private DatabasePopulator databasePopulator;
+
+    @Autowired
+    private HwEntityDao hwEntityDao;
+
+    @Autowired
+    private NodeDao nodeDao;
+
+    @Autowired
+    private AlarmDao alarmDao;
+
+    @Autowired
+    private AlarmDao eventDao;
+
+    @Autowired
+    private MockEventIpcManager eventdIpcMgr;
+
+    private int _id = 1;
+    private MockDatabase mockDatabase;
+    private final NmsInventoryServiceSyncImpl inventoryService = new NmsInventoryServiceSyncImpl();
+
+    @Before
+    public void setUp() throws Exception {
+
+        eventdIpcMgr.setEventWriter(mockDatabase);
+
+        // create data for node
+        populateData();
+
+        // configure server and client
+        configureTestServerWithRealClient();
+    }
+
+    private void populateData() {
+
+        databasePopulator.addExtension(new DatabasePopulator.Extension<HwEntityDao>() {
+
+            @Override
+            public DatabasePopulator.DaoSupport<HwEntityDao> getDaoSupport() {
+                return new DatabasePopulator.DaoSupport<HwEntityDao>(HwEntityDao.class, hwEntityDao);
+            }
+
+            @Override
+            public void onPopulate(DatabasePopulator populator, HwEntityDao dao) {
+                OnmsNode node = databasePopulator.getNode1();
+                OnmsHwEntity port = getHwEntityPort(node);
+                dao.save(port);
+                OnmsHwEntity container = getHwEntityContainer(node);
+                container.addChildEntity(port);
+                dao.save(container);
+                OnmsHwEntity module = getHwEntityModule(node);
+                module.addChildEntity(container);
+                dao.save(module);
+                OnmsHwEntity powerSupply = getHwEntityPowerSupply(node);
+                dao.save(powerSupply);
+                OnmsHwEntity chassis = getHwEntityChassis(node);
+                chassis.addChildEntity(module);
+                chassis.addChildEntity(powerSupply);
+                dao.save(chassis);
+                dao.flush();
+            }
+
+            @Override
+            public void onShutdown(DatabasePopulator populator, HwEntityDao dao) {
+                for (OnmsHwEntity entity : dao.findAll()) {
+                    dao.delete(entity);
+                }
+            }
+        });
+
+        databasePopulator.populateDatabase();
+    }
+
+    private void configureTestServerWithRealClient() throws IOException, InterruptedException, ExecutionException {
+        server = ServerBuilder.forPort(PORT)
```

Comment:
if you are using real client why do you need  `startInProcessChannel()` in the client.